app/controller/List.js

```diff
@@ -82,6 +82,8 @@ Ext.define('WhatsFresh.controller.List', {
 			    },
 			    failure: function() {
 			        console.log('something went wrong!');
+			        WhatsFresh.util.Messages.showLocationError();
+					this.getUseLocationToggle().setValue(0);
```

Comment:
Don't forget that `this` is reassigned on function calls. Since the failure function is nested inside a call to `watchPosition`, we get "undefined is not a function" errors when we go to set the toggle's value.